client/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Route, Redirect } from "react-router-dom";
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+    return (
+        <Route 
+            {...rest}
+            render={() => 
+                localStorage.getItem("token")? (<Component/>) : (<Redirect to="/login"/>)
+            }
```

Comment:
You are redirecting to `/login` here but your `Login` component is rendered at `/` so nothing is rendering when it is redirected here.